packages/edit-site/src/components/list/table.js

```diff
@@ -18,21 +18,50 @@ import { addQueryArgs } from '@wordpress/url';
  */
 import { store as editSiteStore } from '../../store';
 import isTemplateRemovable from '../../utils/is-template-removable';
+import isTemplateRevertable from '../../utils/is-template-revertable';
 
-function Actions( { template, onClose } ) {
-	const { removeTemplate } = useDispatch( editSiteStore );
+function Actions( { template } ) {
+	const { removeTemplate, revertTemplate } = useDispatch( editSiteStore );
+
+	const isRemovable = isTemplateRemovable( template );
+	const isRevertable = isTemplateRevertable( template );
+
+	if ( ! isRemovable && ! isRevertable ) {
+		return null;
+	}
 
 	return (
-		<MenuGroup>
-			<MenuItem
-				onClick={ () => {
-					removeTemplate( template );
-					onClose();
-				} }
-			>
-				{ __( 'Remove template' ) }
-			</MenuItem>
-		</MenuGroup>
+		<DropdownMenu
+			icon={ moreVertical }
+			label={ __( 'Actions' ) }
+			className="edit-site-list-table__actions"
+		>
+			{ ( { onClose } ) => (
+				<MenuGroup>
+					{ isRemovable && (
+						<MenuItem
+							onClick={ () => {
+								removeTemplate( template );
+								onClose();
+							} }
+						>
+							{ __( 'Remove template' ) }
+						</MenuItem>
+					) }
+					{ isRevertable && (
+						<MenuItem
+							info={ __( 'Restore template to theme default' ) }
+							onClick={ () => {
+								revertTemplate( template );
```

Comment:
The main outstanding issue seems to be that `revertTemplate` doesn't actually save the edited entity records to complete the reversion. The listing page doesn't update because of that.

It's really designed for use in the editor.

The options here:
- Don't use this action, create an entirely different action or make the revert code local to this component
- Customize `revertTemplate` so that it accepts additional params to make it work on this screen
- Something else that I haven't thought of

Reply:
We could, not that it's preferred, add the entity saving workflow to this screen too. I think it would look weird though.

Could we add an optional option `force: true` to indicate that it should save immediately after edited? If that doesn't work we can create a separate action for it.

Reply:
I implemented something similar to `force: true` but I didn't like how it means that passing an argument completely changes the behaviour of the action.

Ideally I'd split `revertTemplate` up into two functions—one that saves, one that doesn't—and have them both call into a private action which does the entity editing bits. This is really difficult to do unless we change `edit-site` to use thunks, though. 

So instead I went with e9e7ede831bcdcf8b0fd238455633dbfe19e348a which has the button immediately save the entity after reverting. I added a `allowUndo` option so that the success notice, when we implement notices, doesn't allow the user to undo something that's been already saved.

Reply:
I think we should also inform users that this is an irreversible action.

Reply:
> I think we should also inform users that this is an irreversible action.

I think let's follow up on that. I see "Ability to undo an action" on the list (https://github.com/WordPress/gutenberg/issues/36597). We have the same problem with the _Remove_ action.